### ISSUE TITLE
Simple records count - proof of concept

### ISSUE THREAD

**Issue by LebCit** [olifolkerd/tabulator#3492]
Hello,

Before using Tabulator, I've used other JavaScript grids that had a display count records functionality.
In one of them, the pagination option displays a counter like : "_Showing X to Y of Z results_".
Because I want the same counter in Tabulator, I came up with the following :
```js
function recordsCountDisplay() {

    let myTableInfos = document.getElementById("myTableInfos");

    table.on("tableBuilt", function () {
        // table has finished being rendered.
        let tt = document.querySelector(".tabulator-table"),
            tf = document.querySelector(".tabulator-footer"),
            tp = document.querySelector(".tabulator-paginator");

        myTableInfos.textContent = `Showing 1 to ${tt.childElementCount} of ${table.getDataCount()} results`;

        tf.insertBefore(myTableInfos, tp);
    });

    table.on("pageLoaded", function (pageno) {
        //pageno - the number of the loaded page
        let ttcec = document.querySelector(".tabulator-table").childElementCount,
            tps = table.options.paginationSize,
            tmp = table.getPageMax(),
            fristRowNumber = (pageno * tps) - (tps - 1);
        if (tmp > 1) {
            myTableInfos.textContent = `Showing ${fristRowNumber} to ${fristRowNumber+ttcec-1} of ${table.getDataCount("active")} results`;
        }
    });

    table.on("dataFiltered", function (filters, rows) {
        //filters - array of filters currently applied
        //rows - array of row components that pass the filters
        let tps = table.options.paginationSize;

        if (rows.length < tps) {
            myTableInfos.textContent = `Showing 1 to ${rows.length} of ${rows.length} results`;
        } else {
            myTableInfos.textContent = `Showing 1 to ${tps} of ${rows.length} results`;
        }
    });

}
```
Here is a [working pen](https://codepen.io/LebCit/pen/dyVPwWK) as a proof of concept.
Try to modify the paginationSize option and/or add more items.
I've tried it with a data of more than 100 rows and a lot of filters, it works perfectly.
I'm sure @olifolkerd can do much better by far, but this will do the work if you have a paginated data with filters.

SYA,
LebCit

**Comment by bornova** [comment]
Hi @LebCit,
There was a discussion about this a while ago (#3106) but it went stale.  It would be a nice feature to have, although I am not sure if there can be one solution that fits all.  I tend to agree with this comment: https://github.com/olifolkerd/tabulator/issues/3106#issuecomment-751524775

On that note, there are many ways you can do this on your own, as you have done.  However, I should point out your suggestion won't work with the `paginationSizeSelector` option.  An alternative way to do what you are after may be something like this:
```javascript
let myTableInfos = document.createElement('span') // No need to have this span in your html
myTableInfos.id = 'myTableInfos'

function updateRecordCount () {
  let activeRecords = table.getDataCount('active')
  let pageSize = table.getPageSize()
  let pageNo = table.getPage()

  let from = (pageNo - 1) * pageSize + 1
  let to = pageNo * pageSize > activeRecords || pageSize === true ? activeRecords : pageNo * pageSize

  myTableInfos.innerHTML = 'Showing ' + from + ' to ' + to + ' of ' + activeRecords + ' results'
}

table.on('tableBuilt', function () {
  // table has finished being rendered.
  document.querySelector('.tabulator-footer').prepend(myTableInfos)
})

table.on('pageLoaded', function () {
  updateRecordCount()
})

table.on('dataFiltered', function () {
  setTimeout(updateRecordCount, 10) // Wait for table to update
})
```
https://codepen.io/izmirli/pen/rNGaEyX

**Comment by LebCit** [comment]
Hello @bornova 

Thanks a lot for your response.
Indeed my code was not intended to work with `paginationSizeSelector`, and as mentioned in the comment you refer to :
>Every project would probably want it displayed differently, if displayed at all.
Thanks a lot for sharing. Your solution covers more use case than mine and is much better.

SYA,
LebCit

**Comment by olifolkerd** [comment]
Hey All,

I have just pushed an update to the 5.1 branch that adds the `paginationCounter` option that allows for display of a configurable pagination counter.

This will be included in the 5.1 release.

Cheers

Oli :)
